controllers/clusterversion_controller.go

```diff
@@ -436,3 +464,82 @@ func (c *ClusterVersionReconciler) ensureConsolePlugin() error {
 
 	return nil
 }
+
+func (c *ClusterVersionReconciler) registerWebhook() error {
+	whConfig := &admrv1.ValidatingWebhookConfiguration{}
+	whConfig.Name = "subscription.ocs.openshift.io"
+
+	err := c.createOrUpdate(whConfig, func() error {
+
+		// openshift fills in the ca on finding this annotation
+		whConfig.Annotations = map[string]string{
+			"service.beta.openshift.io/inject-cabundle": "true",
+		}
+
+		// take a copy of the template
+		wh := *templates.SubscriptionValidatingWebhook.DeepCopy()
+		if len(whConfig.Webhooks) != 0 {
+			// do not mutate CA bundle that was injected by openshift
+			wh.ClientConfig.CABundle = whConfig.Webhooks[0].ClientConfig.CABundle
+		}
+
+		// only send requests received from own namespace
+		wh.NamespaceSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"kubernetes.io/metadata.name": c.OperatorNamespace,
+			},
+		}
+
+		// only send resources matching the label
+		wh.ObjectSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				subscriptionLabelKey: subscriptionLabelValue,
+			},
+		}
+
+		// send request to the service running in own namespace
+		wh.ClientConfig.Service.Namespace = c.OperatorNamespace
+
+		// set the webhook
+		whConfig.Webhooks = []admrv1.ValidatingWebhook{wh}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+
+	c.log.Info("successfully registered validating webhook")
+	return nil
+}
+
+func labelClientOperatorSubscription(c *ClusterVersionReconciler) error {
+	subscriptionList := &opv1a1.SubscriptionList{}
+	err := c.List(c.ctx, subscriptionList, client.InNamespace(c.OperatorNamespace))
+	if err != nil {
+		return fmt.Errorf("failed to list subscriptions")
+	}
+
+	sub := utils.Find(subscriptionList.Items, func(sub *opv1a1.Subscription) bool {
+		return sub.Spec.Package == "ocs-client-operator"
+	})
+
+	if sub == nil {
+		return fmt.Errorf("failed to find subscription with ocs-client-operator package")
+	}
+
+	if sub.GetLabels() == nil {
+		sub.Labels = map[string]string{}
+	}
+	if sub.Labels[subscriptionLabelKey] != subscriptionLabelValue {
+		sub.Labels[subscriptionLabelKey] = subscriptionLabelValue
+		err = c.Update(c.ctx, sub)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
We have an `addLabel` util in the storage client controller, can we pull it into a util and use it here?
(https://github.com/red-hat-storage/ocs-client-operator/blob/main/controllers/storageclient_controller.go#L444-L451)

Reply:
had to create new util which takes key & value.
done.